lib/puppet/util/selinux.rb

```diff
@@ -137,6 +142,33 @@ def set_selinux_default_context(file)
     nil
   end
 
+  ##
+  # normalize_selinux_category is an internal, method that converts
+  # all selinux categories to the human-readable representation,
+  # avoiding potential issues when mcstransd is not functional.
+  #
+  # It is not marked private because it is needed by File's
+  # selcontext.rb, but it is not intended for use outside of Puppet's
+  # code.
+  #
+  # @return [String] the human-readable category name
+  def normalize_selinux_category(category)
+    # We don't cache this, but there's already a ton of duplicate work
+    # in the selinux handling code.
+
+    File.open(Selinux.selinux_translations_path).each do |line|
```

Comment:
@vinzent How likely is that file to not exist when selinux is being used? My poking around (admittedly just on a couple of distros) show it seems to come along with the `selinux-policy` packages, so should basically always exist.

If it doesn't exist, we have two options: bail with an exception (the current case), or silently go forward without translations. Going forward silently would essentially be the same behavior as what this patch is trying to solve. I'm hesitant to leave behavior that can cause downstream resources to be refreshed accidentally, and would much rather just bail the catalog.